Sources/xcodeproj/Objects/Targets/PBXReferenceProxy.swift

```diff
@@ -65,30 +56,20 @@ public final class PBXReferenceProxy: PBXObject {
             remoteReference = objectReferenceRepository.getOrCreate(reference: remoteRefString, objects: objects)
         }
         fileType = try container.decodeIfPresent(.fileType)
-        path = try container.decodeIfPresent(.path)
-        sourceTree = try container.decodeIfPresent(.sourceTree)
         try super.init(from: decoder)
     }
-}
-
-// MARK: - PBXReferenceProxy
 
-extension PBXReferenceProxy: PlistSerializable {
-    func plistKeyAndValue(proj _: PBXProj, reference: String) -> (key: CommentedString, value: PlistValue) {
-        var dictionary: [CommentedString: PlistValue] = [:]
+    override func plistKeyAndValue(proj: PBXProj, reference: String) throws -> (key: CommentedString, value: PlistValue) {
+        guard case .dictionary(var dictionary) = try super.plistKeyAndValue(proj: proj, reference: reference).value else {
+            fatalError("super implementation changed and we didn’t realise!")
```

Comment:
Instead of using `fatalError`, I'd add a test that fails if the super implementation changed. With that test in place, we can safely assume that the value that we get here is a dictionary.

Reply:
Do you have an example of how you’d like this to be done?

Reply:
Happy to do this in order to get this merged, just not sure how to go about it without more direction. Thanks.

Reply:
K I figured something out. Coming up.

Reply:
Although even with a test added to detect if the super changes, I’d still have to `fatalError` here unless you have another suggestion.